chips/sam4l/src/wdt.rs

```diff
@@ -0,0 +1,95 @@
+use core::cell::Cell;
+use core::mem;
+use kernel::common::volatile_cell::VolatileCell;
+use kernel::hil;
+use pm::{self, Clock, PBDClock};
+
+#[repr(C, packed)]
+pub struct WdtRegisters {
+    cr: VolatileCell<u32>,
+    clr: VolatileCell<u32>,
+    sr: VolatileCell<u32>,
+    ier: VolatileCell<u32>,
+    idr: VolatileCell<u32>,
+    imr: VolatileCell<u32>,
+    isr: VolatileCell<u32>,
+    icr: VolatileCell<u32>,
+}
+
+// Page 59 of SAM4L data sheet
+const BASE_ADDRESS: *mut WdtRegisters = 0x400F0C00 as *mut WdtRegisters;
+
+pub struct Wdt {
+    registers: *mut WdtRegisters,
+    enabled: Cell<bool>,
+}
+
+pub static mut WDT: Wdt = Wdt::new(BASE_ADDRESS);
+
+impl Wdt {
+    const fn new(base_address: *mut WdtRegisters) -> Wdt {
+        Wdt {
+            registers: base_address,
+            enabled: Cell::new(false),
+        }
+    }
+
+    fn start(&self) {
+        let regs: &mut WdtRegisters = unsafe { mem::transmute(self.registers) };
+
+        self.enabled.set(true);
+
+        unsafe {
+            pm::enable_clock(Clock::PBD(PBDClock::WDT));
+        }
+
+        let control = (1 << 16) | // Clock enable
+                      (17 << 8) | // Set PSEL to 17, ~2s watchdog period
```

Comment:
Should this be configurable through the HIL? If not, at least configurable by in a chip-specific way?